src/plugins/calculationTry1/MathService.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { createContext } from "react";
+
+
+export type THistoryFormulas = {
+  history: string;
+};
+
+
+const ERROR_MESSAGES = Object.freeze({
+  formula: "Проверьте формулу, она некорректна",
+  symbols:
+    "Некорректные символы в формуле!\n Допустимы цифры 0-9, -, +, *, /, (, ), . \n Формула должна начинаться с '='",
+});
+let historyFormulas = { history: "" };
+const MathServiceContext: React.Context<THistoryFormulas> = createContext(null);
+
+function MathService(props: { children: JSX.Element }) {
+  return <MathServiceContext.Provider value={historyFormulas}>{props.children}</MathServiceContext.Provider>;
+}
+
+function MathCounter(formula: string): number | Error {
```

Comment:
Counter - usually used exactly for counter (as i++)

I goes a better name MathCacl|MathCalculator|...evaluator....